engines/hadesch/rooms/monster/illusion.cpp

```diff
@@ -0,0 +1,374 @@
+	_flightShootProjectileFrame = _flightShootAnimFrame + birdInfo[_birdType]._projectileFrame;
+	_flightShootEndFrame = _flightShootAnimFrame + (birdInfo[_birdType]._projectileFrame == 6 ? 13 : 18);
+	_hasShot = false;
+}
+
+// 15201
+void Bird::tick(Common::SharedPtr <Bird> backRef, Common::SharedPtr<Battleground> battleground) {
+	Common::SharedPtr<VideoRoom> room = g_vm->getVideoRoom();
+	if (!_isActive)
+		return;
+	LayerId flyLayer = LayerId(birdInfo[_birdType]._flyAnim, _id, "bird");
+	LayerId shootLayer = LayerId(birdInfo[_birdType]._shootAnim, _id, "bird");
+	_flightCounterMs = (g_vm->getCurrentTime() - _flightStart);
+	if (_flightCounterMs < _flightLengthMs)
+	{
+		int frame = _flightCounterMs / 100;
+		FlightPosition fp = getFlightPosition(_flightCounterMs / (double) _flightLengthMs);
+		int scale = fp.scale;
+		if (frame < _flightShootAnimFrame) {
+			Common::Point cornerPos = fp.centerPos - (scale / 100.0)
+				* birdInfo[_birdType]._birdSize;
+			room->selectFrame(flyLayer, 500, frame % 5, cornerPos);
+			room->setScale(flyLayer, scale);
+			room->stopAnim(shootLayer);
+		} else if (frame < _flightShootEndFrame) {
+			Common::Point cornerPos = fp.centerPos - (scale / 100.0)
+				* birdInfo[_birdType]._birdShootSize;
+			room->selectFrame(shootLayer, 500, frame - _flightShootAnimFrame, cornerPos);
+			room->setScale(shootLayer, scale);
+			room->stopAnim(flyLayer);
+		} else { // 15204
+			Common::Point cornerPos = fp.centerPos - (scale / 100.0)
+				* birdInfo[_birdType]._birdSize;
+			room->selectFrame(flyLayer, 500, (frame - _flightShootEndFrame) % 5, cornerPos);
+			room->setScale(flyLayer, scale);
+			room->stopAnim(shootLayer);
+		}
+
+		if (frame >= _flightShootProjectileFrame && !_hasShot) {
+			_hasShot = true;
+			battleground->launchProjectile(scale / 2, fp.centerPos, _targetPos.x < _startPos.x ? -1 : +1);
+		}
+	}
+	else
+	{
+		room->stopAnim(flyLayer);
+		room->stopAnim(shootLayer);
+		_isActive = false;
+	}
+
+	return;
+}
+
+void Bird::handleAbsoluteClick(Common::Point p) {
+	Common::SharedPtr<VideoRoom> room = g_vm->getVideoRoom();
+	if (!_isActive || _flightCounterMs >= _flightLengthMs)
+		return;
+	FlightPosition fp = getFlightPosition(_flightCounterMs / (double) _flightLengthMs);
+	int r = fp.scale * 40 / 100;
+	if (p.sqrDist(fp.centerPos) > r * r)
+		return;
+	room->stopAnim(LayerId(birdInfo[_birdType]._flyAnim, _id, "bird"));
+	room->stopAnim(LayerId(birdInfo[_birdType]._shootAnim, _id, "bird"));
+	_isActive = false;
+	LayerId l = LayerId(birdInfo[_birdType]._interceptAnim, _id, "bird");
+	room->playAnimWithSound(l, "v7220ec0", 500, PlayAnimParams::disappear(),
+		EventHandlerWrapper(),
+		fp.centerPos - birdInfo[_birdType]._birdSize * (fp.scale / 100.0));
+}
+
+Illusion::Illusion(Common::SharedPtr<Battleground> battleground) {
+	_battleground = battleground;
+	for (unsigned i = 0; i < 3; i++)
+		_birds[i] = Common::SharedPtr<Bird>(new Bird(i));
+}
+
+void Illusion::launchBird() {
+	for (unsigned i = 0; i < 3; i++)
```

Comment:
I think we prefer always using braces if the thing inside the `for` is more than one line

Reply:
Fixed